wire-ios/Wire-iOS/Sources/UserInterface/GroupDetails/GroupDetailsViewController.swift

```diff
@@ -324,6 +337,66 @@ extension GroupDetailsViewController {
 
 }
 
+// MARK: - Conversation verification status
+
+private extension GroupDetailsViewController {
+
+    var verificationStatus: NSAttributedString? {
+        guard conversation.isVerified else {
+            return nil
+        }
+        return attributedString(title: verificationStatusTitle, icon: verificationStatusIcon)
+    }
+
+    var verificationStatusTitle: String {
+        typealias ConversationVerificationStatus = L10n.Localizable.GroupDetails.ConversationVerificationStatus
+
+        switch conversation.messageProtocol {
+        case .proteus, .mixed:
+            return ConversationVerificationStatus.proteus
+        case .mls:
+            return ConversationVerificationStatus.e2ei
+        }
+    }
+
+    var verificationStatusIcon: UIImage {
+        switch conversation.messageProtocol {
+        case .proteus, .mixed:
+            return Asset.Images.verifiedShield.image
+        case .mls:
+            return Asset.Images.certificateValid.image
+        }
+    }
+
+    var verificationStatusColor: UIColor {
+        switch conversation.messageProtocol {
+        case .proteus, .mixed:
+            return SemanticColors.Label.textCertificateVerified
+        case .mls:
+            return SemanticColors.Label.textCertificateValid
+        }
+    }
+
+    func attributedString(title: String, icon: UIImage) -> NSAttributedString {
+        let attributes: [NSAttributedString.Key: Any] = [
+            .foregroundColor: verificationStatusColor
+        ]
+        let attributedString = NSMutableAttributedString(string: title, attributes: attributes)
+        let imageTextAttachment = NSTextAttachment(image: icon)
+
+        let imageSize = CGSize(width: 12, height: 12)
+        let verticalOffset: CGFloat = -2
+        imageTextAttachment.bounds = CGRect(x: 0, y: verticalOffset, width: imageSize.width, height: imageSize.height)
+
+        let imageTextAttachmentString = NSAttributedString(attachment: imageTextAttachment)
+        attributedString.append(" ".attributedString)
+        attributedString.append(imageTextAttachmentString)
+
+        return attributedString
+    }
+
```

Comment:
We have an NSAttachement here can we make sure we call this method in traitCollectionDidChange so we can make sure the icon updates its color from light to dark mode and vice versa?

Reply:
Let's keep it for now and test it during the playlest